spyder/api/widgets/menus.py

```diff
@@ -168,6 +168,19 @@ def add_action(self: T,
         self._dirty = True
         self._actions_map[item_id] = action
 
+    def remove_action(self, item_id: str):
+        if item_id in self._actions_map:
+            action = self._actions_map.pop(item_id)
+            position = None
+
+            for i, (_, act) in enumerate(self._actions):
+                if act == action:
+                    position = i
+                    break
+
+            if position is not None:
+                self._actions.pop(position)
```

Comment:
I think you need to mark the menu as dirty by setting `self._dirty = True` here, so it is re-render the next time a user clicks on it.